### PR TITLE
AUT-2418: Create an API gateway deployed from an OpenAPI spec

### PR DESCRIPTION
**Pull request by whi-tw**
## What

Start migration of account-management to new OpenAPI-based api gateway.

Details in the commit messages.

There are also a couple of drive-bys (marked BAU)

## How to review

- Code review
- Deploy to an authdev, ensure that nothing is deleted when deploying it (other than the API gateway deployment)

### CODE REVIEW

File: ci/terraform/interventions-api-stub/lambda.tf [hunk 1]
```diff
@@ -7,7 +7,7 @@ module "account_interventions_stub_role" {
 }
 
 module "account_interventions_stub_lambda" {
```
Comment (dbes-gds) [inline ci/terraform/interventions-api-stub/lambda.tf:9]: does this need changing?

File: ci/terraform/ticf-cri-stub/lambda.tf [hunk 2]
```diff
@@ -7,7 +7,7 @@ module "ticf_cri_stub_role" {
 }
 
 module "ticf_cri_stub_lambda" {
-  source = "../modules/stub-endpoint-module"
+  source = "../modules/endpoint-lambda"
```
Comment (dbes-gds) [inline ci/terraform/ticf-cri-stub/lambda.tf:10]: needs to change?

File: ci/terraform/interventions-api-stub/lambda.tf (hunk 1 above)
Comment (whi-tw) [reply ci/terraform/interventions-api-stub/lambda.tf:9]: Yup - this new module is a rework of the old stub-endpoint-module (226873160032c33c8708b6d38d173d93af4d1917)

File: ci/terraform/ticf-cri-stub/lambda.tf (hunk 2 above)
Comment (whi-tw) [reply ci/terraform/ticf-cri-stub/lambda.tf:10]: Yup - this new module is a rework of the old stub-endpoint-module (226873160032c33c8708b6d38d173d93af4d1917)

**Review by dbes-gds — APPROVED** [review]

**Pull request merged**